app/src/ui/changes/oversized-files-warning.tsx

```diff
@@ -0,0 +1,104 @@
+import { Dialog, DialogContent, DialogFooter } from '../dialog'
+import { LinkButton } from '../lib/link-button'
+import { Monospaced } from '../lib/monospaced'
+import { PathText } from '../lib/path-text'
+import { Dispatcher } from '../../lib/dispatcher'
+import { Repository } from '../../models/repository'
+import { ICommitContext } from '../../models/commit'
+
+const GitLFSWebsiteURL =
+  'https://help.github.com/articles/versioning-large-files/'
+
+interface IOversizedFilesProps {
+  readonly oversizedFiles: ReadonlyArray<string>
+  readonly onDismissed: () => void
+  readonly dispatcher: Dispatcher
+  readonly context: ICommitContext
+  readonly repository: Repository
+}
+
+/** A dialog to display a list of files that are too large to commit. */
+export class OversizedFiles extends React.Component<IOversizedFilesProps> {
+  private closeButton: Button | null = null
+
+  public constructor(props: IOversizedFilesProps) {
+    super(props)
+  }
+
+  private onCloseButtonRef = (button: Button | null) => {
+    this.closeButton = button
+  }
+
+  public componentDidMount() {
+    // Since focus is given to the Git LFS link by default, we will instead set focus onto the cancel button.
+    if (this.closeButton != null) {
+      this.closeButton.focus()
+    }
+  }
+
+  public render() {
+    return (
+      <Dialog
+        id="oversized-files"
+        title={__DARWIN__ ? 'Files Too Large' : 'Files too large'}
+        onDismissed={this.props.onDismissed}
+        type="warning"
+      >
+        <DialogContent>
+          <p>
+            The following files are over 100MB.{' '}
+            <strong>
+              If you commit these files, you will no longer be able to push this
+              repository to GitHub.com.
+            </strong>
+          </p>
+          {this.renderFileList()}
+          <p className="recommendation">
+            We recommend you avoid committing these files or use{' '}
+            <LinkButton uri={GitLFSWebsiteURL}>Git LFS</LinkButton> to store
+            large files on GitHub.
+          </p>
+        </DialogContent>
+
+        <DialogFooter>
+          <ButtonGroup destructive={true}>
+            <Button type="submit" ref={this.onCloseButtonRef}>
+              Cancel
+            </Button>
+            <Button onClick={this.commitAnyway}>
+              {__DARWIN__ ? 'Commit Anyway' : 'Commit anyway'}
+            </Button>
+          </ButtonGroup>
+        </DialogFooter>
+      </Dialog>
+    )
+  }
+
+  private renderFileList() {
+    return (
+      <div className="files-list">
+        {this.props.oversizedFiles.map(fileName => (
+          <ul key={fileName}>
+            <Monospaced>
+              <PathText path={fileName} />
+            </Monospaced>
+          </ul>
+        ))}
+      </div>
+    )
+  }
+
+  private commitAnyway = async () => {
+    const context = this.props.context
+    this.props.dispatcher.commitIncludedChanges(this.props.repository, context)
```

Comment:
```suggestion
    await this.props.dispatcher.commitIncludedChanges(this.props.repository, context)
```

Reply:
i _think_ we don't want to clear the message until the commit is finished. is that right?

Reply:
@outofambit 
I can certainly make this change. The reason I chose not to await `commitIncludedChanges` is because creating the commit may take a while, especially with a lot of files. It seemed preferable to get the popup out of the way since it was no longer necessary and allow them to see the commit creation now they normally would.

Perhaps we should show a spinning wheel on the Commit Anyway button to indicate that it's working on it? Otherwise I would be fine with just adding the `await`, I'd just like to mention this to see what would be the ideal way for the user.

Reply:
To add to this:
> i think we don't want to clear the message until the commit is finished. is that right?

That is correct, if we clear the commit message while it is committing, the commit summary will display the default commit message for the selected files.

However, we can change that so that it does not get cleared until after the commit. We can solve that regardless of whether or not we want to close the pop-up while committing. If we want to keep the current functionality of closing the popup during the commit, we can just move the `closePopup()` to happen before the await `commitIncludedChanges`.

Reply:
> we can just move the closePopup() to happen before the await commitIncludedChanges.

that's a great suggestion. let's do that! @Daniel-McCarthy :D